### ISSUE TITLE
User import (ldap) 1102 (with 10.0.12/13/14) instead of 5394 (with 10.0.11)

### ISSUE THREAD

**Issue by TheRoby** [glpi-project/glpi#16725]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Version

10.0.13-dev

### Bug description

Hello,

I have a surprising functioning for LDAP import 

In 10.0.11 5394 users imported
```
 5394/5394 [============================] 100%
+--------------+---------+-------------+--------------------------+----------------------------------+
| Serveur LDAP | Importé | Synchronisé | Supprimé du serveur LDAP | Restauré depuis un annuaire LDAP |
+--------------+---------+-------------+--------------------------+----------------------------------+
| 4            | 0       | 5394        | 0                        | 0                                |
+--------------+---------+-------------+--------------------------+----------------------------------+

```

In 10.0.12 (and .13-dev)  1102 users imported
```
 1102/1102 [============================] 100%
+--------------+---------+-------------+--------------------------+----------------------------------+
| Serveur LDAP | Importé | Synchronisé | Supprimé du serveur LDAP | Restauré depuis un annuaire LDAP |
+--------------+---------+-------------+--------------------------+----------------------------------+
| 4            | 0       | 1102        | 0                        | 0                                |
+--------------+---------+-------------+--------------------------+----------------------------------+
```

if I put the src/User.php file from version 10.0.11, then it works as before

As a result, all of my users are not updated. I haven't managed to find any common point that could explain why only some people go back.

The problem is that I use the Last Synchronization field in users...

Thank you :)

**Comment by TheRoby** [comment]
Hi,
Same problem with 10.0.13

**Comment by TheRoby** [comment]
https://github.com/glpi-project/glpi/commit/2ee311a00acf2753bf57a81e4a04e482de4a47de
if i change this :
```
public function getFromDBbyDn($user_dn)
    {
       [old] return $this->getFromDBByCrit(['user_dn' => $user_dn]);
        /**
         * We use the 'user_dn_hash' field instead of 'user_dn' for performance reasons.
         * The 'user_dn_hash' field is a hashed version of the 'user_dn' field
         * and is indexed in the database, making it faster to search.
         */
        return $this->getFromDBByCrit([
            'user_dn_hash' => md5($user_dn)
        ]);
    }
```
to the old version, I have 5394 records updated ...

**Comment by TheRoby** [comment]
I think I found a clue :
https://github.com/glpi-project/glpi/commit/2ee311a00acf2753bf57a81e4a04e482de4a47de#commitcomment-139834247

**Comment by cedric-anne** [comment]
Hi,

This issue will be fixed by #16806. As it requires a migration, the patch cannot be applied in a production environment.

You can apply the following patch instead, on GLPI 10.0.14, to fix the issue until GLPI 10.0.15 is released:
```diff 
diff --git a/src/User.php b/src/User.php
index 695fe4df5f..5349a49322 100644
--- a/src/User.php
+++ b/src/User.php
@@ -567,7 +567,7 @@ class User extends CommonDBTM
          * and is indexed in the database, making it faster to search.
          */
         return $this->getFromDBByCrit([
-            'user_dn_hash' => md5($user_dn)
+            'user_dn_hash' => md5(Sanitizer::sanitize($user_dn))
         ]);
     }
 
@@ -847,7 +847,7 @@ class User extends CommonDBTM
     {
         // Hash user_dn if set
         if (isset($this->input['user_dn']) && is_string($this->input['user_dn']) && strlen($this->input['user_dn']) > 0) {
-            $this->input['user_dn_hash'] = md5($this->input['user_dn']);
+            $this->input['user_dn_hash'] = md5(Sanitizer::sanitize($this->input['user_dn']));
         }
     }
 
@@ -3503,7 +3503,7 @@ HTML;
         // Hash user_dn if is updated
         if (in_array('user_dn', $this->updates)) {
             $this->updates[] = 'user_dn_hash';
-            $this->fields['user_dn_hash'] = is_string($this->input['user_dn']) && strlen($this->input['user_dn']) > 0 ? md5($this->input['user_dn']) : null;
+            $this->fields['user_dn_hash'] = is_string($this->input['user_dn']) && strlen($this->input['user_dn']) > 0 ? md5(Sanitizer::sanitize($this->input['user_dn'])) : null;
         }
     }
 
```
